src/compiler/scala/tools/nsc/io/Jar.scala

```diff
@@ -77,6 +77,7 @@ class Jar(file: File) extends AbstractIterable[JarEntry] {
     Iterator continually in.getNextJarEntry() takeWhile (_ != null) foreach f
   }
   override def iterator: Iterator[JarEntry] = this.toList.iterator
+  override def isEmpty: Boolean = iterator.isEmpty
```

Comment:
I've left `def isEmpty: Boolean = iterator.isEmpty` in some subclasses (which is the same as `super.isEmpty`) in thie PR to highlight to spots where there is no clean direct way to implement it. I could remove them before merge if desired.

Reply:
@retronym Couldn't "def iterator: Iterator[JarEntry]" be `Iterator continually in.getNextJarEntry() takeWhile (_ != null)`

And then "def foreach" is `iterator.foreach`?

(Then we don't need to create the full list to test isEmpty)

Reply:
Yep, we could do something smarter here. However, I added the comment to this line not because `JarEntry.isEmpty` is interesting performance wise, but just because it was the first instance of the few cases where it was either not possible or not worth the effort to write a custom `isEmpty` implementation.